File: Java/Numbers/PrimeFactors/PrimeFactors.java
import java.util.*;

/**
 * Created by sraone on 10/5/14.
 */
public class PrimeFactors {
    /**
     * Gets the input from the user to print fibonacci series of those many numbers
     * @return
     */
    public static int get_input(){
        System.out.print("Enter the number you want to find prime factors for ... ");
        return (Integer.parseInt(System.console().readLine()));
    }

    /**
     * Returns all the prime factors of the given number
     * @param number
     * @return
     */
    public static List <Integer> get_factors (int number){
        List<Integer> factors = new ArrayList<Integer>();
        for (int i = 1; i <= number; i++ ){
            if (number % i == 0 ) factors.add(i);
        }
        return factors;
    }

    /**
     * Finds the given number is a prime number or not
     * @param args
     */
    public static boolean is_prime(int num){
        for (int i = 2; i<num; i++) {
            if (num % i == 0 ) return false;
        }
        return true;
    }

    // Main Method
    public static void main(String[] args) {
        int num = get_input();
        System.out.print("Prime factors for " + num + " are ... ");
        for (int i : get_factors(num)){
            if (is_prime(i)) System.out.print(i + " ");
        }
        System.out.println();
    }
}

File: Python/Numbers/PrimeFactors.py
'''
Prime Factorization - Have the user enter a number and find all Prime Factors (if there are any) and display them.

Created on 10 Oct 2014

@author: sraone
'''
def get_factors (num):
    temp = list()
    for i in range(2,num) :
        if num%i == 0:
            temp.append(i)
    return temp

def is_prime (number):
    if not get_factors(number): return True
    else: False

def get_primefactors ():
    number = int (input('Enter the number ... '))
    pf = list()
    for i in get_factors(number):
        if (is_prime(i)): 
            pf.append(i)
    return pf

if __name__ == '__main__':
    print(get_primefactors())
    